### PR TITLE
Take a tag reference instead of a hash

### PR DESCRIPTION
**Pull request by ps-jay**
It would be useful to be able to specify tag names instead of hash ID's
to kick off builds.

https://trello.com/c/cs4TkIdT

### CODE REVIEW

**Comment by ps-jay** [comment]
Updated to "git reference" as per disucssion in gitter

**Comment by ps-jay** [comment]
Okay - it's ready to go IMHO !

File: dockci/models/build.py [hunk 1]
```diff
@@ -155,8 +154,6 @@ def validate(self):
 
             if not self.job:
                 errors.append("Parent job not given")
-            if self.commit and not is_git_hash(self.commit):
-                errors.append("Invalid git commit hash")
```
Comment (ps-jay) [inline dockci/models/build.py]: Can't confirm this until after git_info stage has run...

File: dockci/templates/job.html [hunk 2]
```diff
@@ -12,12 +12,12 @@
   <input id="filterStable" type="checkbox">Only stable releases
 </label></div>
 <table id="builds-list" class="table table-striped">
-  <tr><th>Created</th><th>Commit</th><th>Version</th><th>Author</th><th>State</th></tr>
+  <tr><th>Created</th><th>Tag</th><th>Commit ref</th><th>Author</th><th>State</th></tr>
```
Comment (ps-jay) [inline dockci/templates/job.html]: Looks better with tag to the left of full-hashes IMHO

File: dockci/views/build.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import logging
 import mimetypes
 import os.path
-import re
+# import re
```
Comment (RickyCook) [inline dockci/views/build.py]: Commented import won't be needed

Comment (ps-jay) [inline dockci/views/build.py]: Done

**Pull request merged**